src/LodeRunner.API.Test/IntegrationTests/Controllers/Clients.cs

```diff
@@ -77,17 +93,33 @@ public async Task CanGetEndpointsReturnSuccessAndCorrectContentType(string url,
         /// <returns><see cref="Task"/> representing the asynchronous unit test.</returns>
         [Fact]
         [Trait("Category", "Integration")]
-        private async Task CanGetClients()
+        public async Task CanGetClients()
         {
             using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
-
             using var l8rService = await ComponentsFactory.CreateAndStartLodeRunnerServiceInstance(nameof(this.CanGetClients));
-
             string clientStatusId = l8rService.ClientStatusId;
 
-            Assert.False(string.IsNullOrEmpty(clientStatusId), "Unable to retrieve ClientStatusId.");
+            this.output.WriteLine($"Started LodeRunner (client mode) [ClientStatusId: {clientStatusId}]");
+
+            Assert.False(string.IsNullOrEmpty(clientStatusId), "Unable to retrieve ClientStatusId from LodeRunner (client mode) service.");
 
-            await httpClient.WaitAndValidateGetClientsToMatchId(ClientsUri, clientStatusId, this.jsonOptions, this.output);
+            HttpResponseMessage httpResponse = await httpClient.GetAsync($"{ClientsUri}");
+
+            if (httpResponse.StatusCode == HttpStatusCode.OK || httpResponse.StatusCode == HttpStatusCode.NoContent)
+            {
+                this.output.WriteLine($"UTC Time:{DateTime.UtcNow}\tAction: GET all Client\tResponse StatusCode: '{httpResponse.StatusCode}'");
+            }
+            else
+            {
+                this.output.WriteLine($"UTC Time:{DateTime.UtcNow}\tAction: GET all Client\tUNEXPECTED Response StatusCode: '{httpResponse.StatusCode}'");
+            }
+
+            List<Client> clients = await httpResponse.Content.ReadFromJsonAsync<List<Client>>(this.jsonOptions);
+
+            Assert.Equal(HttpStatusCode.OK, httpResponse.StatusCode);
+            Assert.NotEmpty(clients);
+
```

Comment:
It looks like we expect to have clients and a status code of 200, but in line 108 we test for 200 and 204 and indicate that it is okay.  If the two asserts are for 200 and content then the conditional above it should only be testing for 200.  Otherwise, if the condition is correct then the asserts would have to change, but that would violate the idea that the test should be deterministic.

Reply:
After discussion, it was to accommodate an unknown state due to the order in testing.  We'll keep it like this for the moment as we have a plan to move away from this testing and drive the API externally.

Reply:
Moving the log output to the `HttpClientExtension` so the logic doesn't conflict with the `Assert()` in the test method.